esmvalcore/preprocessor/__init__.py

```diff
@@ -363,7 +360,12 @@ def _get_multi_model_settings(products, step):
     return settings, exclude
 
 
-def _run_preproc_function(function, items, kwargs, input_files=None):
+def _run_preproc_function(
+    function: Callable,
+    items: Any,
```

Comment:
```suggestion
    items: PreprocessorFile | Cube | str | Path | Iterable[PreprocessorFile], Iterable[Cube], Iterable[str], Iterable[Path],
```
and perhaps create an [alias](https://docs.python.org/3/library/typing.html#type-aliases) for these to improve readability?